src/OrchardCore.Modules/OrchardCore.Media.AmazonS3/CreateMediaS3BucketEvent.cs

```diff
@@ -84,4 +85,39 @@ public override async Task ActivatingAsync()
             }
         }
     }
+
+    public override async Task RemovingAsync(ShellRemovingContext context)
+    {
+        if (_options.RemoveBucket && !String.IsNullOrEmpty(_options.BucketName))
+        {
+            try
+            {
+                var isBucketExists = await AmazonS3Util.DoesS3BucketExistV2Async(_amazonS3Client, _options.BucketName);
+                if (!isBucketExists)
+                {
+                    return;
+                }
+
+                var bucketRequest = new DeleteBucketRequest
+                {
+                    BucketName = _options.BucketName,
+                    UseClientRegion = true
+                };
+
+                // Trying to create bucket.
+                var response = await _amazonS3Client.DeleteBucketAsync(bucketRequest);
+                if (!response.IsSuccessful())
+                {
+                    _logger.LogError("Unable to remove the Amazon S3 Bucket {BucketName}", _options.BucketName);
+                    context.ErrorMessage = $"Failed to remove the Amazon S3 Bucket '{_options.BucketName}'.";
+                }
+            }
+            catch (Exception ex)
```

Comment:
Can't we use a more specific exception type? Or check with `IsFatal()`.

Reply:
Idem, same pattern as in `ActivatingAsync()`, if so we may need to change both, as you want

Reply:
And same pattern in `CreateMediaBlobContainerEvent` for Azure blob.

For azure blob the methods specify more specific exception types.

But for AmazonS3 the methods don't specify any exception types.

Maybe @neglectedvalue, who I think did the amazonS3 module, has some suggestions?

Reply:
Okay done, refactored both files and both `Activating()` and `Removing()` in each file.

Reply:
We can use AmazonS3Exception here, or create more specific one inherited from AmazonS3Exception.
But I guess AmazonS3Exception is enough.

Reply:
@neglectedvalue Okay cool thanks!